### PR TITLE
Remove forest-crypto crate

### LINKED ISSUE #2641: Remove the `forest_crypto` crate
**Issue summary**

The `forest_crypto` trait contains a `Signer` trait which is unused and a `VRFProof` structure (which only deals with json-formatting). Let's move the `VRFProof` structure to either `forest_utils` or `forest_json` and then delete the `forest_crypto` crate.

**Other information and links**

### PR DESCRIPTION
**Pull request by creativcoder**
## Summary of changes

Changes introduced in this pull request:
- Removes forest-crypto crate and moves relevant items to forest-json crate.

## Reference issue to close (if applicable)

Closes #2641 

## Change checklist

- [x] I have performed a self-review of my own code,
- [X] I have made corresponding changes to the documentation,
- [X] I have added tests that prove my fix is effective or that my feature works (if possible),
- [X] I have made sure the [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

File: forest/daemon/src/daemon.rs [hunk 1]
```diff
@@ -93,6 +93,7 @@ pub(super) async fn start(opts: CliOpts, config: Config) -> anyhow::Result<Db> {
             let file = write_to_file(&gen_keypair.encode(), &path, "keypair")?;
             // Restrict permissions on files containing private keys
             forest_utils::io::set_user_perm(&file)?;
+            #[allow(deprecated)] // FIXME: use: into_ed25519
```
Comment (LesnyRumcajs) [inline forest/daemon/src/daemon.rs:96]: Can we do it in this PR? If not, let's include a tracking issue in the comment. Wild FIXMEs tend to get forgotten.

File: node/forest_libp2p/src/service.rs [hunk 2]
```diff
@@ -849,6 +849,7 @@ pub fn get_keypair(path: &Path) -> Option<Keypair> {
         Ok(mut vec) => match ed25519::Keypair::decode(&mut vec) {
             Ok(kp) => {
                 info!("Recovered libp2p keypair from {:?}", &path);
+                #[allow(deprecated)] // FIXME: use: into_ed25519
```
Comment (LesnyRumcajs) [inline node/forest_libp2p/src/service.rs:852]: Can we do it in this PR? If not, let's include a tracking issue in the comment. Wild FIXMEs tend to get forgotten.

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

File: node/forest_libp2p/src/service.rs (hunk 2 above)
Comment (creativcoder) [reply node/forest_libp2p/src/service.rs]: Added in this PR.

File: forest/daemon/src/daemon.rs (hunk 1 above)
Comment (creativcoder) [reply forest/daemon/src/daemon.rs]: Added in this PR.

File: forest/daemon/src/daemon.rs [hunk 3]
```diff
@@ -87,13 +85,21 @@ pub(super) async fn start(opts: CliOpts, config: Config) -> anyhow::Result<Db> {
     let net_keypair = match get_keypair(&path.join("keypair")) {
         Some(keypair) => Ok::<forest_libp2p::Keypair, std::io::Error>(keypair),
         None => {
-            let gen_keypair = ed25519::Keypair::generate();
+            let gen_keypair = forest_libp2p::Keypair::generate_ed25519();
```
Comment (lemmih) [inline forest/daemon/src/daemon.rs:88]: Sigh, it's silly that they don't have a way to convert `ed25519::Keypair` into ` Keypair`. Oh well.

File: node/forest_libp2p/src/behaviour.rs [hunk 4]
```diff
@@ -5,12 +5,13 @@ use ahash::{HashMap, HashSet};
...
 use forest_libp2p_bitswap::BitswapBehaviour;
 use libp2p::{
-    core::{identity::Keypair, PeerId},
+    core::identity::Keypair,
     gossipsub::{
         self, IdentTopic as Topic, MessageAuthenticity, MessageId, PublishError, SubscriptionError,
         ValidationMode,
     },
     identify,
+    identity::PeerId,
```
Comment (lemmih) [inline node/forest_libp2p/src/behaviour.rs:14]:
```suggestion
    identity::{self, PeerId, Keypair},
```

File: forest/daemon/src/daemon.rs (hunk 3 above)
Comment (lemmih) [reply forest/daemon/src/daemon.rs:88]: Could you open an issue with libp2p and describe our use-case?

**Review by lemmih — CHANGES_REQUESTED** [review]

File: forest/daemon/src/daemon.rs (hunk 3 above)
Comment (creativcoder) [reply forest/daemon/src/daemon.rs:88]: https://github.com/libp2p/rust-libp2p/issues/3618

File: node/forest_libp2p/src/behaviour.rs (hunk 4 above)
Comment (creativcoder) [reply node/forest_libp2p/src/behaviour.rs]: added

**Review by lemmih — APPROVED** [review]

File: node/forest_libp2p/src/behaviour.rs (hunk 4 above)
Comment (lemmih) [reply node/forest_libp2p/src/behaviour.rs:14]: Pretty sure you can do it like I showed.

Comment (lemmih) [reply node/forest_libp2p/src/behaviour.rs:14]: (except for the `identify` module. i missed that.)

**Pull request merged**